components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/db/SitePermissionEntity.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions.db
+
+import android.arch.persistence.room.ColumnInfo
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.PrimaryKey
+import mozilla.components.feature.sitepermissions.SitePermissions
+
+/**
+ * Internal entity representing a site permission as it gets saved to the database.
+ */
+@Entity(tableName = "site_permissions")
+internal data class SitePermissionsEntity(
+
+    @PrimaryKey
+    @ColumnInfo(name = "origin")
+    var origin: String,
+
+    @ColumnInfo(name = "location")
+    var location: Boolean,
+
+    @ColumnInfo(name = "notification")
+    var notification: Boolean,
+
+    @ColumnInfo(name = "microphone")
+    var microphone: Boolean,
+
+    @ColumnInfo(name = "camera_back")
+    var cameraBack: Boolean,
+
+    @ColumnInfo(name = "camera_front")
+    var cameraFront: Boolean,
+
+    @ColumnInfo(name = "bluetooth")
+    var bluetooth: Boolean,
+
+    @ColumnInfo(name = "local_storage")
+    var localStorage: Boolean,
```

Comment:
Like above: I don't think `Boolean`s are good enough here. Maybe an `enum` instead? (You'll need a type converter for Room in that case, I guess).